test/cql/lwt_test.result

```diff
@@ -1760,9 +1760,17 @@ Error from server: code=2200 [Invalid query] message="Invalid null map key"
 | True        | [False] |
 +-------------+---------+
 > update lwt set c = [false] where a = 1 if c = [null];
-Error from server: code=2200 [Invalid query] message="null is not supported inside collections"
++-------------+---------+
+| [applied]   | c       |
+|-------------+---------|
+| False       | [False] |
++-------------+---------+
```

Comment:
As you said in the commit message, this suddenly became legal. Do we want it to become legal? Is it only legal now in LWT? I thought you recently had a patch which explicitly said that we use a list with null in it internally (to implement IN for LWT) but it is not allowed in real data. So what does "c = [null]" mean?

Reply:
> As you said in the commit message, this suddenly became legal. Do we want it to become legal? 

I do

> Is it only legal now in LWT? 

It's now legal everywhere we fully switched to evaluate().

> I thought you recently had a patch which explicitly said that we use a list with null in it internally (to implement IN for LWT) but it is not allowed in real data. So what does "c = [null]" mean?

If checks if c is equal to a list with one element, which is NULL. Since c cannot contain nulls, the result is false, but asking the question whether those two lists is legal.

Reply:
Regarding your last comment - it is true, but do we also allow "x = 3" when x is a string column? We could claim to justify allowing that query that this comparison is legal and always false (a string can never be a number). But it's more useful to tell the user that they probably had a typo in their query than to silently return nothing. In some sense `[null]` is the same.
 
On the other hand, maybe you have some "parametric code" in mind which the user can plug in any bound variable, and you want to allow null on the bound variable and do nothing instead of erroring out?

Reply:
`x = 3` is a type error. It's wrong for all values of x and for all values of 3, if x is a text column.

`x = :var` is not a type error. `x` and `:var` have exactly the same type. A good type system allows all legal values of a type to be used, so that the type check is sufficient and necessary (a counterexample is division, which doesn't accept a zero divisor, but it's generally regarded has bad).

We could disallow `x = [null]` and allow `x = :var`, but it's just another inconsistency. For example, it will lead to different results when using prepared statements and simple statements. How do I check my prepared statement in cqlsh?

There are more than forty years of experience in SQL. We should take advantage of them, and of our users' experience in them, instead of trying to reinvent little corners of SQL here or there. Even if we get something better (and I don't think we will), we lose in surprising the users.